tests/test_providers.py

```diff
@@ -515,6 +515,13 @@ def test_get_instance_name(
     ) == expected.format(inode=tmp_path.stat().st_ino)
 
 
+@pytest.mark.parametrize(
+    "cache_path",
+    [
+        None,
+        pathlib.Path("/tmp/charmcraft-cache"),
```

Comment:
This global cache may affect other test runs in the same machine, shouldn't we remove it after running this specific test with all parametrized test cases?